internal/extramanifest/extramanifest_test.go

```diff
@@ -0,0 +1,184 @@
+	cms := []*corev1.ConfigMap{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "extra-manifest-cm1",
+				Namespace: "default",
+			},
+			Data: map[string]string{
+				"sriovnetwork1.yaml": sriovnetwork1,
+				"sriovnetwork2.yaml": sriovnetwork2,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "extra-manifest-cm2",
+				Namespace: "default",
+			},
+			Data: map[string]string{
+				"sriovnodepolicy1.yaml": sriovnodepolicy1,
+				"sriovnodepolicy2.yaml": sriovnodepolicy2,
+			},
+		},
+	}
+
+	for _, cm := range cms {
+		err = fakeClient.Create(context.Background(), cm)
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+	}
+
+	handler := &EMHandler{
+		Client: fakeClient,
+		Log:    ctrl.Log.WithName("ExtraManifest"),
+	}
+
+	// Export the manifests to the temporary directory
+	err = handler.ExportExtraManifestToDir(context.Background(),
+		[]lcav1alpha1.ConfigMapRef{
+			{Name: "extra-manifest-cm1", Namespace: "default"},
+			{Name: "extra-manifest-cm2", Namespace: "default"},
+		}, toDir)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// Check that the manifests were exported to the correct files
+	expectedFilePaths := []string{
+		filepath.Join(toDir, extraManifestPath, "0_sriov-nw-mh_openshift-sriov-network-operator.yaml"),
+		filepath.Join(toDir, extraManifestPath, "0_sriov-nw-fh_openshift-sriov-network-operator.yaml"),
+		filepath.Join(toDir, extraManifestPath, "1_sriov-nnp-mh_openshift-sriov-network-operator.yaml"),
+		filepath.Join(toDir, extraManifestPath, "1_sriov-nnp-fh_openshift-sriov-network-operator.yaml"),
+	}
+
+	for _, expectedFile := range expectedFilePaths {
+		if _, err := os.Stat(expectedFile); os.IsNotExist(err) {
+			t.Fatalf("Expected file %s does not exist", expectedFile)
+		}
+	}
+
+	// Apply the manifests that were previously exported
+	err = handler.ApplyExtraManifestsFromDir(context.Background(), toDir)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// Check that the manifests were applied
+	sriovnetworks := &unstructured.UnstructuredList{}
+	sriovnetworks.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "sriovnetwork.openshift.io",
+		Version: "v1",
+		Kind:    "SriovNetworkList",
+	})
+
+	err = fakeClient.List(context.Background(), sriovnetworks)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	if len(sriovnetworks.Items) != 2 {
+		t.Errorf("Expected 2 SriovNetworks, got %d", len(sriovnetworks.Items))
+	}
+
+	sriovnodepolicies := &unstructured.UnstructuredList{}
+	sriovnodepolicies.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "sriovnetwork.openshift.io",
+		Version: "v1",
+		Kind:    "SriovNetworkNodePolicyList",
+	})
+
+	err = fakeClient.List(context.Background(), sriovnodepolicies)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	if len(sriovnodepolicies.Items) != 2 {
+		t.Errorf("Expected 2 SriovNetworkNodePolicies, got %d", len(sriovnodepolicies.Items))
```

Comment:
not very important but we already have added `testify` package. You can rely on assert for `if` statement and the expect and got messages.

Reply:
Yeah, right. Somehow I don't know why I didn't use that. Perhaps I thought I only have a few checks.